commit 31426b50594c12802e77bcba69f5f5169cdf0dd5
Author: Lubomir Rintel <lkundrak@v3.sk>
Date:   Mon Sep 24 13:18:34 2018 +0100

    Revert "uapi/linux/keyctl.h: don't use C++ reserved keyword as a struct member name"
    
    commit 8c0f9f5b309d627182d5da72a69246f58bde1026 upstream.
    
    This changes UAPI, breaking iwd and libell:
    
      ell/key.c: In function 'kernel_dh_compute':
      ell/key.c:205:38: error: 'struct keyctl_dh_params' has no member named 'private'; did you mean 'dh_private'?
        struct keyctl_dh_params params = { .private = private,
                                            ^~~~~~~
                                            dh_private
    
    This reverts commit 8a2336e549d385bb0b46880435b411df8d8200e8.
    
    Fixes: 8a2336e549d3 ("uapi/linux/keyctl.h: don't use C++ reserved keyword as a struct member name")
    Signed-off-by: Lubomir Rintel <lkundrak@v3.sk>
    Signed-off-by: David Howells <dhowells@redhat.com>
    cc: Randy Dunlap <rdunlap@infradead.org>
    cc: Mat Martineau <mathew.j.martineau@linux.intel.com>
    cc: Stephan Mueller <smueller@chronox.de>
    cc: James Morris <jmorris@namei.org>
    cc: "Serge E. Hallyn" <serge@hallyn.com>
    cc: Mat Martineau <mathew.j.martineau@linux.intel.com>
    cc: Andrew Morton <akpm@linux-foundation.org>
    cc: Linus Torvalds <torvalds@linux-foundation.org>
    cc: <stable@vger.kernel.org>
    Signed-off-by: James Morris <james.morris@microsoft.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/include/uapi/linux/keyctl.h b/include/uapi/linux/keyctl.h
index 910cc4334b21..7b8c9e19bad1 100644
--- a/include/uapi/linux/keyctl.h
+++ b/include/uapi/linux/keyctl.h
@@ -65,7 +65,7 @@
 
 /* keyctl structures */
 struct keyctl_dh_params {
-	__s32 dh_private;
+	__s32 private;
 	__s32 prime;
 	__s32 base;
 };
diff --git a/security/keys/dh.c b/security/keys/dh.c
index 1a68d27e72b4..b203f7758f97 100644
--- a/security/keys/dh.c
+++ b/security/keys/dh.c
@@ -300,7 +300,7 @@ long __keyctl_dh_compute(struct keyctl_dh_params __user *params,
 	}
 	dh_inputs.g_size = dlen;
 
-	dlen = dh_data_from_key(pcopy.dh_private, &dh_inputs.key);
+	dlen = dh_data_from_key(pcopy.private, &dh_inputs.key);
 	if (dlen < 0) {
 		ret = dlen;
 		goto out2;
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1054_31426b50594c12802e77bcba69f5f5169cdf0dd5.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1054_31426b50594c12802e77bcba69f5f5169cdf0dd5.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
